ts/data.ts

```diff
@@ -1,8 +1,26 @@
 /* exported data */
 
-let data = {
+interface Data {
+  view: string;
+  entries: Obj[];
+  editing: null;
+  nextEntryId: number;
+}
+
+let data: Data = {
   view: 'entry-form',
   entries: [],
   editing: null,
   nextEntryId: 1,
 };
+
+window.addEventListener('beforeunload', () => {
+  const dataJSON = JSON.stringify(data);
+  localStorage.setItem('data-model', dataJSON);
+});
+
+if (localStorage.getItem('data-model')) {
+  const storedData = localStorage.getItem('data-model');
+  if (!storedData) throw new Error('The $storedData query has failed.');
```

Comment:
No need to throw an error here. If there's nothing in local storage, then we just want to use the initial value we assigned to it